File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProf =0;
        for(int i=0; i<prices.size(); i++){
            for(int j=i+1; j<prices.size(); j++){
                int tempProf = prices[j] - prices[i];
                if(tempProf>maxProf){
                    maxProf = tempProf;
                }
            }
        }
        return maxProf;
    }
};

File: Data Structures & Algorithms/ipo/submission-0.py
class Solution:
    def findMaximizedCapital(self, k: int, w: int, profits: List[int], capital: List[int]) -> int:
        maxProf = []
        minCap = [(c,p) for c,p in zip(capital, profits)]
        heapq.heapify(minCap)
        for i in range(k):
            while minCap and minCap[0][0]<=w:
                c,p = heapq.heappop(minCap)
                heapq.heappush(maxProf, -p)
            if not maxProf:
                break
            p = -heapq.heappop(maxProf)
            w+=p
        return w